.github/workflows/ci.yaml

```
@@ -255,8 +255,15 @@ jobs:
           ./sbt -jvm-opts autotest/.jvmopts "project autotest" \
             coverageReport
 
+      # To save significant time on file upload, we first tar up the directory
+      # as it contains > 10k files and the artefact upload was taking almost
+      # 1hr.
+      - name: Package Coverage Report
+        run: |
+          tar cvf coverage_report.tar autotest/target/coverage-report/
```

:thinking: if upload is where the script is taking most of the hit, it may be worth adding some additional compression to reduce the size

```suggestion
          tar cvzf coverage_report.tar.gz autotest/target/coverage-report/
```

---

I felt it was more the IO of opening and closing all the files - especially seeing a lot are tiny. Plus, the artefact upload action compresses to zip as well.